File: 04_sum_of_n_in_range/04_sum_of_no_in_range.dart
import 'dart:io';

// program to find sum of range with provide starting and ending values

int summation(int a, int b){
  int result = 0;
  for(int i = 0; i <= b; i++){
    result = result + i;
  }
  return result;
}

void main(){
  try{
    stdout.write("Enter starting range : ");
    int a = int.parse(stdin.readLineSync()!);
    stdout.write("Enter ending range : ");
    int b = int.parse(stdin.readLineSync()!);

    if (a < 0 || b < 0){
      print("Enter positive integer value");
    }
    else{
      int result = summation(a, b);
      print("Sum of range between $a and $b is : $result");
    }

  }
  on FormatException catch (e){
    print("Invalid input : $e");
  }
  catch (e){
    print("Exception caught : $e");
  }
}

File: 04_sum_of_n_in_range/04_sum_of_no_in_range.py
# python program to find sum of numbers in provided range
# method 1 - traditional way
try:
    m,n=map(int,input().split(','))
    sum=0
    for i in range(m,n+1):
        sum=sum+i
    print(sum)
except:
    print("Enter valid range i.e. from starting to ending n,m")

# method 2 - using recurssion
try:
    m,n=map(int,input().split(','))
    def sum(m,n):
        if m>n:
            return 0
        summ=m+sum(m+1,n)
        return summ
    print(sum(m,n))
except:
    print("Enter valid range i.e. from starting to ending n,m")
